cpp/src/arrow/compute/exec/expression_benchmark.cc

```diff
@@ -69,6 +70,26 @@ static void SimplifyFilterWithGuarantee(benchmark::State& state, Expression filt
   }
 }
 
+static void ExecuteScalarExpressionOverhead(benchmark::State& state, Expression expr) {
+  const auto rows_per_batch = static_cast<int32_t>(state.range(0));
+  const auto num_batches = 10000000 / rows_per_batch;
```

Comment:
Why so many batches? Does it make the benchmark more useful?

Reply:
I basically wanted to benchmark the execution with different batch sizes but with the same total data amount. So, if we  want to benchmark on batches with 10 rows each, then we should have 100000 batches, thus in total we will have 10000000 data points. Keeping the total data fix, the benchmarking tries to see how the execution gets affected when we execute the same amount of data but with batches of different sizes.